lib/puppet/provider/iis_application_pool/webadministration.rb

```diff
@@ -28,10 +28,17 @@ def update
     Puppet.debug "Updating #{@resource[:name]}"
     cmd = []
 
-    @resource.properties.select{|rp| rp.name != :ensure && rp.name != :state }.each do |property|
+    @resource.properties.select{|rp| rp.name != :ensure && rp.name != :state}.each do |property|
       property_name = iis_properties[property.name.to_s]
-      Puppet.debug "Changing #{property_name} to #{property.value}"
-      cmd << "Set-ItemProperty -Path 'IIS:\\AppPools\\#{@resource[:name]}' -Name '#{property_name}' -Value #{property.value};"
+      if @resource[:name].is_a?(Hash)
```

Comment:
Yes, abstracting elements based on type is better than hard-coding.
But it needs to be: `if property.value.is_a?(Array)` and `property.value.each do |item|`
Should I add a commit or rebase with these changes?